src/Tasks/Microsoft.NET.Build.Tasks/FindItemsFromPackages.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using NuGet.Packaging.Core;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Returns the ITaskItem values in Items that were resolved from the specified
+    /// set of Packages.
+    /// </summary>
+    /// <remarks>
+    /// Both Items and Packages are expected to have 'PackageName' and 'PackageVersion' 
+    /// metadata properties to use for the matching.
+    /// </remarks>
+    public sealed class FindItemsFromPackages : TaskBase
+    {
+        private readonly List<ITaskItem> _itemsFromPackages = new List<ITaskItem>();
+
+        [Required]
+        public ITaskItem[] Items { get; set; }
+
+        [Required]
+        public ITaskItem[] Packages { get; set; }
+
+        [Output]
+        public ITaskItem[] ItemsFromPackages
+        {
+            get { return _itemsFromPackages.ToArray(); }
```

Comment:
I'll fix this in the new files I'm adding.  Note that this is a pattern used in about half the places this repo.